irksome/stepper.py

```diff
@@ -1,14 +1,82 @@
 from .getForm import getForm, AI
 from firedrake import NonlinearVariationalProblem as NLVP
 from firedrake import NonlinearVariationalSolver as NLVS
-from firedrake import Function, norm
 from firedrake.dmhooks import pop_parent, push_parent
 import numpy
+from .stage import StageValueTimeStepper
 
 
-class TimeStepper:
+def TimeStepper(F, butcher_tableau, t, dt, u0, bcs=None,
+                solver_parameters=None,
+                update_solver_parameters=None,
+                nullspace=None,
+                stage_type="deriv", appctx=None,
+                bc_type=None, splitting=None):
+    """Helper function to dispatch between various back-end classes
+       for doing time stepping.  Returns an instance of the
+       appropriate class.
+
+    :arg F: A :class:`ufl.Form` instance describing the semi-discrete problem
+            F(t, u; v) == 0, where `u` is the unknown
+            :class:`firedrake.Function and `v` iss the
+            :class:firedrake.TestFunction`.
+    :arg butcher_tableau: A :class:`ButcherTableau` instance giving
+            the Runge-Kutta method to be used for time marching.
+    :arg t: A :class:`firedrake.Constant` instance that always
+            contains the time value at the beginning of a time step
+    :arg dt: A :class:`firedrake.Constant` containing the size of the
+            current time step.  The user may adjust this value between
+            time steps, but see :class:`AdaptiveTimeStepper` for a
```

Comment:
Probably remove comment about `AdaptiveTimeStepper` for now

Reply:
Also below in docstring for `StageDerivativeTimeStepper`

Reply:
Yes, good catch.